src/components/lib/kernel/cos_kernel_api.c

```diff
@@ -742,45 +734,167 @@ __alloc_mem_cap(struct cos_compinfo *ci, cap_t ct, vaddr_t *kmem, capid_t *cap)
 	return 0;
 }
 
-static thdcap_t
-__cos_thd_alloc(struct cos_compinfo *ci, compcap_t comp, thdclosure_index_t init_data)
+/*
```

Comment:
I can't really really follow the next functions, what they're doing, and if they can be replaced by the composer vas functionality in conjunction with the *general* APIs with checks behind them to do the "right thing". I'm really just ignorant here, and can't reverse engineer this here.